core/src/main/java/hudson/model/ManagementLink.java

```diff
@@ -214,5 +215,15 @@ public enum Category {
         }
     }
 
+    /**
+     * A badge shown as overlay over the icon on "Manage Jenkins".
+     *
+     * @return badge or {@code null} if no badge should be shown.
+     * @since TODO
+     */
+    public @CheckForNull Badge getBadge() {
```

Comment:
One drawback of the new `Badge` type is that it requires a new core dependency. Previously, the `String` based API could be implemented even by plugins with an old core dependency.

I don't think this change must be reverted, but considering simple page navigation shouldn't be expensive anyway, the argument for one call instead of two isn't _that_ strong (expensive computation should be done elsewhere and cached values shown, kind of like available updates).